starport_template/lib/widgets/wallets_list_sheet.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:cosmos_ui_components/components/empty_list_message.dart';
+import 'package:cosmos_ui_components/cosmos_theme.dart';
+import 'package:cosmos_ui_components/cosmos_ui_components.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_mobx/flutter_mobx.dart';
+import 'package:starport_template/starport_app.dart';
+import 'package:transaction_signing_gateway/model/wallet_public_info.dart';
+
+class WalletsListSheet extends StatefulWidget {
+  const WalletsListSheet({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  _WalletsListSheetState createState() => _WalletsListSheetState();
+}
+
+class _WalletsListSheetState extends State<WalletsListSheet> {
+  List<WalletPublicInfo> get publicInfos => StarportApp.walletsStore.wallets;
+
+  WalletPublicInfo get selectedWallet => StarportApp.walletsStore.selectedWallet;
+
+  List<WalletInfo> get walletInfos => publicInfos
+      .map(
+        (publicInfo) => WalletInfo(
+          name: publicInfo.name,
+          address: publicInfo.publicAddress,
+          walletId: publicInfo.walletId,
+        ),
+      )
+      .toList();
+
+  @override
+  Widget build(BuildContext context) {
+    //ignore: deprecated_member_use_from_same_package
+    return CosmosBottomSheetContainer(
+      child: SafeArea(
+        child: Observer(
+          builder: (context) => ContentStateSwitcher(
+            emptyChild: const EmptyListMessage(
+              message: "No wallets found. Add one.",
+            ),
+            isEmpty: walletInfos.isEmpty,
+            contentChild: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                SizedBox(height: CosmosTheme.of(context).spacingM),
+                _buildTopActions(),
+                Expanded(
+                  child: Padding(
+                    padding: EdgeInsets.symmetric(horizontal: CosmosTheme.of(context).spacingM),
+                    child: Column(
+                      children: [
+                        SizedBox(height: CosmosTheme.of(context).spacingL),
+                        const CosmosBottomSheetHeader(title: 'Accounts'),
```

Comment:
as mentioned in the `CosmosBottomSheetHeader`, you can make it accept the `leading` and `actions` parameters to simplify the UI

Reply:
Updated